java/cdk-environment/src/main/java/sleeper/environment/cdk/buildec2/BuildEC2Stack.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.environment.cdk.buildec2;
+
+import sleeper.environment.cdk.util.AppContext;
+import sleeper.environment.cdk.util.MyIpUtil;
+import software.amazon.awscdk.CfnOutput;
+import software.amazon.awscdk.NestedStack;
+import software.amazon.awscdk.services.ec2.*;
+import software.constructs.Construct;
+
+import java.security.KeyPair;
+import java.util.Collections;
+import java.util.UUID;
+
+public class BuildEC2Stack extends NestedStack {
+
+    public BuildEC2Stack(Construct scope, IVpc vpc, AppContext app) {
+        super(scope, "BuildEC2");
+        BuildEC2Params params = BuildEC2Params.from(app);
+
+        SecurityGroup allowSsh = SecurityGroup.Builder.create(this, "AllowSsh")
+                .vpc(vpc)
+                .description("Allow SSH inbound traffic")
+                .allowAllOutbound(true)
+                .build();
+        allowSsh.addIngressRule(Peer.ipv4(MyIpUtil.findMyIp() + "/32"), Port.tcp(22));
+
+        // Create a new SSH key every time the CDK is run
+        KeyPair keyPair = KeyPairUtil.generate();
+        KeyPairUtil.writePrivateToFile(keyPair, "BuildEC2.pem");
+        CfnKeyPair key = CfnKeyPair.Builder.create(this, "KeyPair")
+                .keyName("BuildEC2-" + UUID.randomUUID())
```

Comment:
It might be useful to let the user specify a name to use here rather than UUID.randomUUID() so that they can tell which key is theirs in the console.

Reply:
I used UUID.randomUUID() because if you set a consistent key name it tries to update the keypair and fails. You can only update a keypair by deleting and recreating it, and the library doesn't do that for some reason.  I couldn't find another Java library. Maybe in order to delete it, it would need to unset it on the EC2 instance so there's no reference to it.

I'll add a comment to explain this.

Reply:
I'll still add an identifier to the name related to the stack and the specific instance of the stack when you deploy multiple instances.

Reply:
OK.

Reply:
Done